### DISCUSSION TITLE
Pre-loading a texture

### DISCUSSION THREAD

**Discussion by Paril** [mikke89/RmlUi#664] [Q&A]
I've got some very large 4k key art that I want to have precached before the menu gets shown (so I can hide the loading behind things that happen before menus are shown) but I can't seem to find some way in Rml to force it to load a given texture. The only thing I can seemingly do is directly call `TextureDatabase::Fetch` and then call `GetDimensions` to force it to load. Is that a correct approach or is there a better way?

**Comment by mikke89** [comment]
You're right, we don't have an official way to do this. I think that's a fine way to do it. 

In RmlUi 6, the equivalent would be:
```cpp
context->GetRenderManager().LoadTexture(path).GetDimensions();
```

In the future, it would be nice to have a way to control these things, such as: pre-fetch, fetch-on-load, and an opt-in for automatic garbage collection for specific textures.
